platform/c2000/inverter_drivers/pmicspidriver.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of the stm32-sine project.
+ *
+ * Copyright (C) 2022 Bernd Ocklin <bernd@ocklin.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "c2000/pmicspidriver.h"
+
+#include "device.h"
+
+uint32_t PmicSpiDriver::m_base = 0;
+
+void PmicSpiDriver::InitGPIO(uint16_t pin, uint32_t cfg) 
+{
+    GPIO_setMasterCore(pin, GPIO_CORE_CPU1);
+    GPIO_setPinConfig(cfg);
+    GPIO_setPadConfig(pin, GPIO_PIN_TYPE_PULLUP);
+    GPIO_setQualificationMode(pin, GPIO_QUAL_ASYNC);
+}
+
+/**
+ * \brief Initialise the SPI port's GPIO lines
+ */
+void PmicSpiDriver::InitGPIOs() {
+
+    m_base = DEVICE_TESLAM3_PMIC_SPI;
+
+    // SPISOMIA.
+    InitGPIO(DEVICE_TESLAM3_GPIO_PIN_PMIC_MISO, DEVICE_TESLAM3_GPIO_CFG_PMIC_MISO);
```

Comment:
No strong feelings. The defaults here are hard coded in silicon FWIW. I suspect that when/if we split things to run on CPU2 this will need to be revisited. Leave it as you have it.